src/api/app/controllers/webui/comments_controller.rb

```diff
@@ -105,12 +106,45 @@ def preview
     end
   end
 
+  def moderate
+    authorize @comment, :moderate?
+
+    state = ActiveModel::Type::Boolean.new.cast(params[:moderation_state])
+
+    status = if @comment.moderate(state)
+               flash.now[:success] = 'Comment moderated successfully.'
+               :ok
+             else
+               flash.now[:error] = "Failed to moderate comment: #{@comment.errors.full_messages.to_sentence}."
+               :unprocessable_entity
+             end
+
+    if Flipper.enabled?(:request_show_redesign, User.session) && ['BsRequest', 'BsRequestAction'].include?(@comment.commentable_type)
+      render(partial: 'webui/comment/beta/comments_thread',
+             locals: { comment: @comment.root, commentable: @comment.commentable, level: 1 },
+             status: status)
+    else
+      render(partial: 'webui/comment/comment_list',
+             locals: { commentable: @comment.commentable, diff_ref: @comment.root.diff_ref },
+             status: status,
+             root_comment: @comment.root)
+    end
+  end
+
   private
 
   def permitted_params
     params.require(:comment).permit(:body, :parent_id, :diff_ref)
   end
 
+  def set_comment
+    @comment = Comment.find(params[:id] || params[:comment_id])
+    return if @comment.present?
+
+    flash[:error] = "Comment with id '#{params[:id]}' doesn't exist."
```

Comment:
Why don't we use the rails way of dealing with nonexistent resources in the controller?

Reply:
Yes